File: hspasov-strawberries/strawberries.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {

	int k, l, r;
	// vector< vector<bool> > isRottenNewDay;

	int goodStrawberries = 0;

	cin >> k >> l >> r;

	vector < vector <bool> >* isRottenNewDay;
	
	vector < vector <bool> >* isRottenPrevDay;

	int strawberryCol, strawberryRow;

	isRottenPrevDay = new vector < vector <bool> >(k, vector<bool>(l, false));

	isRottenNewDay = new vector < vector <bool> >(*isRottenPrevDay);

	while (cin >> strawberryRow >> strawberryCol) {
		(*isRottenPrevDay)[strawberryRow - 1][strawberryCol - 1] = true;
	}

	for (int i = 0; i < r; i++) {

		delete isRottenNewDay;
		isRottenNewDay = new vector < vector <bool> >(*isRottenPrevDay);

		for (int row = 0; row < (*isRottenPrevDay).size(); row++) {

			for (int col = 0; col < (*isRottenPrevDay)[row].size(); col++) {

				if ((*isRottenPrevDay)[row][col] == true) {
					if (row > 0) {
						(*isRottenNewDay)[row - 1][col] = true;
					}
					if (row < (*isRottenPrevDay).size() - 1) {
						(*isRottenNewDay)[row + 1][col] = true;
					}
					if (col > 0) {
						(*isRottenNewDay)[row][col - 1] = true;
					}
					if (col < (*isRottenPrevDay)[row].size() - 1) {
						(*isRottenNewDay)[row][col + 1] = true;
					}
				}
			}
		}		
		delete isRottenPrevDay;
		isRottenPrevDay = new vector < vector <bool> >(*isRottenNewDay);
	}

	for (int row = 0; row < (*isRottenPrevDay).size(); row++) {

		for (int col = 0; col < (*isRottenPrevDay)[row].size(); col++) {

			if ((*isRottenPrevDay)[row][col] == false) {
				goodStrawberries++;
			}
		}
	}

	delete isRottenPrevDay;
	delete isRottenNewDay;

	cout << goodStrawberries << endl;
	return 0;
}

File: hspasov-strawberries/strawberries.js
let k, l, r;

let goodStrawberries = 0;

// get k, l, r
k = 8;
l = 10;
r = 2;

let isRottenCurrentDay = [];
let isRottenNextDay = [];

for (let row = 0; row < k; row++) {

  isRottenCurrentDay.push(new Array(l).fill(false));
  isRottenNextDay.push(new Array(l).fill(false));
}

// get rotten strawberries
const initRottenStrawberries = [
  {
    row: 4,
    col: 8
  },
  {
    row: 2,
    col: 7
  }
];

for (let initRottenStrawberry of initRottenStrawberries) {
  isRottenCurrentDay[initRottenStrawberry.row - 1][initRottenStrawberry.col - 1] = true;
}


for (let day = 0; day < r; day++) {

  for (let row = 0; row < k; row++) {
    
    for (let col = 0; col < l; col++) {
      
      if (isRottenCurrentDay[row][col] === true) {
        
        if (row > 0) {
          isRottenNextDay[row - 1][col] = true;
        }
        if (row < k - 1) {
          isRottenNextDay[row + 1][col] = true;
        }
        if (col > 0) {
          isRottenNextDay[row][col - 1] = true;
        }
        if (col < l - 1) {
          isRottenNextDay[row][col + 1] = true;
        }
      }
    }
  }

  for (let row = 0; row < k; row++) {
    
    for (let col = 0; col < l; col++) {
      isRottenCurrentDay[row][col] = isRottenNextDay[row][col];
    }
  }
}

for (let row = 0; row < k; row++) {

  for (let col = 0; col < l; col++) {

    if (isRottenCurrentDay[row][col] === false) {
      goodStrawberries++;
    }
  }
}

console.log(goodStrawberries);